chroma-common/blockdevices/blockdevice_zfs.py

```diff
@@ -95,14 +105,50 @@ def __exit__(self, type, value, traceback):
             self.unlock_pool()
 
     def lock_pool(self):
-        lock = self.import_locks[self.pool_path]
-        lock.acquire()
-        # log.debug('ZfsDevice acquired lock on %s' % self.pool_path)
+        """
+        Attempt to acquire a lock on a given zpool through a lockfile, increment the reference count each time
+        this method is called, regardless of whether acquire is actually called. Keep trying until we know
+        we are locking the zpool.
+
+        Lock acquire polls the timeout at intervals of LOCK_ACQUIRE_TIMEOUT.
+        """
+        if ZfsDevice.locks_dir_initialized == False:
+            # ensure lock directory exists
+            try:
+                os.mkdir(self.ZPOOL_LOCK_DIR)
+            except OSError:
+                pass
+
+            ZfsDevice.locks_dir_initialized = True
+
+        while not self.lock.i_am_locking():
+            try:
+                self.lock.acquire(self.LOCK_ACQUIRE_TIMEOUT)
+                with open(self.lock.lock_file, 'w') as f:
+                    f.writelines(str(self.lock.pid))
```

Comment:
Should we catch file I/O exceptions (i.e. disk full, etc.) here or is the caller expected to catch those?